web-console/src/lib/compositions/streaming/inspection/useTableUpdater.ts

```diff
@@ -29,15 +29,33 @@ const updateRowsIntegral = (oldRows: Map<number, Row>, deltaRows: Row[]) => {
   return oldRows
 }
 
-const computeAndSet = (prev: Map<number, Row>) => (setRows: Dispatch<SetStateAction<Row[]>>, parsedRows: Row[]) => {
-  updateRowsIntegral(prev, parsedRows)
-  setRows(Array.from(prev.values()).sort((a, b) => a.genId - b.genId))
-  return prev
+const computeAndSet = (integral: Map<number, Row>) => (setRows: Dispatch<SetStateAction<Row[]>>, parsedRows: Row[]) => {
+  setRows(old => {
+    const relationChanged = !old.length
+    if (relationChanged) {
+      integral.clear()
+    }
+    updateRowsIntegral(integral, parsedRows)
+    return Array.from(integral.values()).sort((a, b) => a.genId - b.genId)
+  })
+  return integral
 }
 
-const computeInBackground = (prev: Map<number, Row>) => (_: Dispatch<SetStateAction<Row[]>>, parsedRows: Row[]) => {
-  return updateRowsIntegral(prev, parsedRows)
-}
+const computeInBackground =
+  (integral: Map<number, Row>) => (setRows: Dispatch<SetStateAction<Row[]>>, parsedRows: Row[]) => {
+    setRows(old => {
+      const relationChanged = !old.length
+      if (relationChanged) {
+        integral.clear()
+      }
+      updateRowsIntegral(integral, parsedRows)
+      if (!relationChanged) {
+        return old
+      }
+      return Array.from(integral.values()).sort((a, b) => a.genId - b.genId)
```

Comment:
It's not clear to me why it needs to be sorted? (comment would help)

Reply:
I am not sure either - it is the way it was initially, see the initial code in https://github.com/feldera/feldera/commit/b4038cccea968f4c53f59e1f059a78b5db4c40eb , line 90